controllers/managedocs_controller.go

```diff
@@ -478,6 +478,16 @@ func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
 			if err := r.delete(r.managedOCS); err != nil {
 				return ctrl.Result{}, fmt.Errorf("unable to delete managedocs: %v", err)
 			}
+			// Refreshing local managedOCS object after deletion is scheduled
+			// to avoid conflict while updating status
+			if err := r.get(r.managedOCS); err != nil {
+				if errors.IsNotFound(err) {
+					r.Log.Info("ManagedOCS resource not found")
+					return ctrl.Result{}, nil
+				} else {
+					return ctrl.Result{}, err
+				}
+			}
```

Comment:
Why the early return if no error? what if we would like to add more logic at the end of this function?
Also the message should be a warning and we need more context in the message itself.

```suggestion
			if err := r.get(r.managedOCS); err != nil &&  {
				if !errors.IsNotFound(err) {
					return ctrl.Result{}, err
				}
				r.Log.V(-1).Info("Trying to reload ManagedOCS resource after delete failed, ManagedOCS resource not found")
			}
```

Reply:
Fixed in next iteration